nifi-nar-bundles/nifi-mqtt-bundle/nifi-mqtt-processors/src/main/java/org/apache/nifi/processors/mqtt/adapters/HiveMqV5ClientAdapter.java

```diff
@@ -0,0 +1,194 @@
+        if (!cleanSession) {
+            connectBuilder.sessionExpiryInterval(connectionProperties.getSessionExpiryInterval());
+        }
+
+        final String lastWillTopic = connectionProperties.getLastWillTopic();
+        if (lastWillTopic != null) {
+            connectBuilder.willPublish()
+                    .topic(lastWillTopic)
+                    .payload(connectionProperties.getLastWillMessage().getBytes())
+                    .retain(connectionProperties.getLastWillRetain())
+                    .qos(MqttQos.fromCode(connectionProperties.getLastWillQOS()))
+                    .applyWillPublish();
+        }
+
+        // checking for presence of password because username can be null
+        final char[] password = connectionProperties.getPassword();
+        if (password != null) {
+            connectBuilder.simpleAuth()
+                    .username(connectionProperties.getUsername())
+                    .password(toBytes(password))
+                    .applySimpleAuth();
+
+            clearSensitive(connectionProperties.getPassword());
+            clearSensitive(password);
+        }
+
+        final Mqtt5Connect mqtt5Connect = connectBuilder.build();
+        mqtt5Client.toBlocking().connect(mqtt5Connect);
+    }
+
+    @Override
+    public void disconnect(long disconnectTimeout) throws NifiMqttException {
+        // Currently it is not possible to set timeout for disconnect with HiveMQ Client. (Only connect timeout exists.)
+        mqtt5Client.toBlocking().disconnect();
+    }
+
+    @Override
+    public void close() throws NifiMqttException {
+        // there is no paho's close equivalent in hivemq client
+    }
+
+    @Override
+    public void publish(String topic, NifiMqttMessage message) throws NifiMqttException {
+        mqtt5Client.toAsync().publishWith()
+                .topic(topic)
+                .payload(message.getPayload())
+                .retain(message.isRetained())
+                .qos(Objects.requireNonNull(MqttQos.fromCode(message.getQos())))
+                .send();
+    }
+
+    @Override
+    public void subscribe(String topicFilter, int qos) {
+        Objects.requireNonNull(callback, "callback should be set");
+
+        mqtt5Client.toAsync().subscribeWith()
+                .topicFilter(topicFilter)
+                .qos(Objects.requireNonNull(MqttQos.fromCode(qos)))
+                .callback(mqtt5Publish -> {
+                    final NifiMqttMessage nifiMqttMessage = new NifiMqttMessage();
+                    nifiMqttMessage.setPayload(mqtt5Publish.getPayloadAsBytes());
+                    nifiMqttMessage.setQos(mqtt5Publish.getQos().getCode());
+                    nifiMqttMessage.setRetained(mqtt5Publish.isRetain());
+                    try {
+                        callback.messageArrived(mqtt5Publish.getTopic().toString(), nifiMqttMessage);
+                    } catch (Exception e) {
+                        throw new NifiMqttException(e);
```

Comment:
Exceptions should include a basic message instead of just wrapping the cause. In this case, something like `Arrived message processing failed` would be helpful, include the `topicFilter` would also be useful for troubleshooting.

Reply:
I moved this catch to ConsumeMQTT, this way, the same exception will be thrown for both clients. Is it ok that way?

Reply:
That sounds good, thanks!